src/any_agent/tracing/trace.py

```diff
@@ -148,6 +149,29 @@ class AgentTrace(BaseModel):
 
     model_config = ConfigDict(arbitrary_types_allowed=True)
 
+    @property
+    def execution_time(self) -> timedelta:
+        """Returns the execution time of the AGENT span named 'any_agent' as a datetime.timedelta object.
+
+        The duration is computed from the span's start and end time (in nanoseconds).
+        This functionality relies on the trace behavior where we create a span named 'any_agent' for each agent run,
+        which is defined in each agent framework's `run` function.
+
+        Raises ValueError if there are no spans, if the AGENT span is not found, or if start/end times are missing.
+        """
+        if not self.spans:
+            msg = "No spans found in trace"
+            raise ValueError(msg)
+        for span in self.spans:
+            if span.attributes.get("any_agent.run_id"):
```

Comment:
Maybe we could propose `gen_ai.run_id` or `gen_ai.session_id` as part of the otel genai semantic conventions, to mark the top level span for a request to an agent that is explicitly finished in some way (and not continued ad infinitum).